### PR TITLE
Save .more messages to a database

### LINKED ISSUE #299: Save .more messages to a database
The dictionary `phenny.messages` should be turned into a SQL table and mapped to disk.

### PR DESCRIPTION
**Pull request by Androbin**
Fixes #299
Towards #292

### CODE REVIEW

File: modules/more.py [hunk 1]
```diff
@@ -4,10 +4,23 @@
...
 def setup(self):
-    self.messages = {}
+    self.more_db = db_path(self, 'more')
+
+    connection = sqlite3.connect(self.more_db)
+    cursor = connection.cursor()
+
+    cursor.execute('''create table if not exists more (
+        target        varchar(255),
+        message       varchar(''' + MAX_MSG_LEN + ''')
```
Comment (sushain97) [inline modules/more.py]: Use %-syntax or {}

File: modules/more.py [hunk 2]
```diff
@@ -47,30 +61,36 @@ def more(phenny, input):
...
     target = target.casefold()
-    remaining = len(phenny.messages[target])
 
-    if count > remaining:
-        count = remaining
+    with DatabaseCursor(self.more_db) as cursor:
+        cursor.execute("SELECT message, identifier FROM more WHERE target=? SORT BY identifier ASC LIMIT ?", (target, count))
+        rows = cursor.fetchall()
+
+        cursor.executemany("DELETE FROM more WHERE identifier=?", [(row[1],) for row in rows])
```
Comment (sushain97) [inline modules/more.py]: Seems like `WHERE identifier IN ...` would make more sense here

File: modules/more.py (hunk 1 above)
Comment (sushain97) [reply modules/more.py]: Also, capitalize the keywords please

File: modules/more.py [hunk 3]
```diff
@@ -4,10 +4,23 @@
...
-    self.messages = {}
+    self.more_db = db_path(self, 'more')
+
+    connection = sqlite3.connect(self.more_db)
+    cursor = connection.cursor()
+
+    cursor.execute('''create table if not exists more (
+        target        varchar(255),
+        message       varchar(''' + MAX_MSG_LEN + ''')
+        identifier    integer primary key autoincrement
```
Comment (sushain97) [inline modules/more.py]: `identifier`? why not the standard `id`? also, this should go first

File: modules/posted.py [hunk 4]
```diff
@@ -3,65 +3,56 @@
...
+    connection = sqlite3.connect(self.posted_db)
+    cursor = connection.cursor()
 
-    c.close()
-    conn.close()
+    cursor.execute('''create table if not exists posted (
+        channel    varchar(255),
+        nick       varchar(255),
+        url        varchar(512),
+        time       timestamp date default (datetime('now', 'localtime'))
```
Comment (sushain97) [inline modules/posted.py]: capitalize keywords here

File: tools.py [hunk 5]
```diff
@@ -31,6 +32,23 @@ def setup(self):
     if hasattr(self.config, 'MAX_MSG_LEN'):
         MAX_MSG_LEN = self.config.MAX_MSG_LEN
 
+def db_path(self, name):
+    filename = self.nick + '-' + self.config.host + '.' + name '.db'
+    return os.path.join(os.path.expanduser('~/.phenny'), filename)
+
+class DatabaseCursor():
```
Comment (sushain97) [inline tools.py]: Huh, does python not have one of these built in? maybe in `contextlib`?

File: modules/more.py (hunk 1 above)
Comment (Androbin) [reply modules/more.py]: Other modules didn't capitalize the table creation either.
I just wanted to preserve style.
But I will just go ahead and update them, too.

File: tools.py (hunk 5 above)
Comment (Androbin) [reply tools.py]: Well, the `connection` is a `contextmanager`, but the `cursor` is not.

File: modules/more.py (hunk 1 above)
Comment (sushain97) [reply modules/more.py]: Hm, don't worry about other modules in this PR but let's make this one proper.

**Comment by sushain97** [comment]
Build?

**Comment by Androbin** [comment]
Sorry, syntax issue.

**Comment by Androbin** [comment]
Okay, seems like I messed up quite a few things.

**Comment by Androbin** [comment]
Imagine me realizing that the database wasn't reset after each test or even between runs.

**Comment by Androbin** [comment]
@sushain97 Build is passing now, including all 17 `more` tests.

**Comment by sushain97** [comment]
Looks pretty good to me. Can I see some screenshots of it in action as well?

**Comment by shardulc** [comment]
@Androbin waiting on this for screenshots.

**Comment by Androbin** [comment]
screenshot

**Comment by Androbin** [comment]
* Had to fix some leftovers from #275
* Screenshot shows expected behavior
* Length of "(N remaining)" is not accounted for during queuing

**Comment by sushain97** [comment]
Build doesn't look too happy anymore.

**Comment by Androbin** [comment]
Yeah, but only the `nightly` build and because of `test_slogan`.

**Comment by Androbin** [comment]
There was a un-related test failure on the `nightly` build. I have forced Travis to re-build.

**Comment by Androbin** [comment]
@shardulc Are you happy with these changes?

**Comment by shardulc** [comment]
@Androbin The screenshot shows "(2" on one line and then "remaining)" on the next line. This is likely something to do with `break_up` function that was changed somewhere in these changes. Can you fix it?

**Comment by Androbin** [comment]
@shardulc See 4 comments above yours.

**Comment by shardulc** [comment]
@Androbin Sorry, missed that. Looks good to me. I'm just verifying the tests, give me a minute.

**Pull request merged**